integration/helpers.go

```diff
@@ -222,26 +221,33 @@ func AssertMatchesCommandError(t testing.TB, expected, actual error) {
 	}
 }
 
-// AssertMatchesWriteError asserts error codes are the same.
-//
-// TODO check not only code; make is look similar to AssertMatchesCommandError above.
-// https://github.com/FerretDB/FerretDB/issues/2545
+// AssertMatchesWriteError asserts that both errors are equal WriteErrors,
+// except messages (and ignoring the Raw part).
+// WriteErrors are wrapped in a WriteException, it asserts there is exactly one WriteError.
 func AssertMatchesWriteError(t testing.TB, expected, actual error) {
 	t.Helper()
 
-	var aErr, eErr mongo.WriteException
+	a, okA := actual.(mongo.WriteException) //nolint:errorlint // do not inspect error chain
+	require.Truef(t, okA, "actual is %T, not mongo.WriteException", a)
+	require.Lenf(t, a.WriteErrors, 1, "actual is %v, expected one mongo.WriteError", a.WriteErrors)
 
-	if ok := errors.As(actual, &aErr); !ok || len(aErr.WriteErrors) != 1 {
-		assert.Equal(t, expected, actual)
-		return
-	}
+	e, okE := expected.(mongo.WriteException) //nolint:errorlint // do not inspect error chain
 
-	if ok := errors.As(expected, &eErr); !ok || len(eErr.WriteErrors) != 1 {
-		assert.Equal(t, expected, actual)
-		return
-	}
+	require.Truef(t, okE, "expected is %T, not mongo.WriteException", e)
+	require.Lenf(t, e.WriteErrors, 1, "expected is %v, expected one mongo.WriteError", e.WriteErrors)
+
+	aErr := a.WriteErrors[0]
+	eErr := e.WriteErrors[0]
 
-	assert.Equal(t, eErr.WriteErrors[0].Code, aErr.WriteErrors[0].Code)
+	aErr.Raw = nil
+	eErr.Raw = nil
+
+	actualMessage := aErr.Message
+	aErr.Message = eErr.Message
+
+	if !AssertEqualWriteError(t, eErr, aErr) {
```

Comment:
So, `AssertEqualWriteError` is similar to `AssertMatchesWriteError`, but requires the same error message? So, "matches" means the same code but possible different messages? Are we fine with the naming of these functions? Should `AssertMatchesWriteError` be called something like `AssertEqualWriteErrorCodes` or something like this for more consistency?

Reply:
Just looking at `WriteError` https://github.com/mongodb/mongo-go-driver/blob/master/mongo/errors.go#L302, we don't ignore `Index` and `Details` so it compares them too. :hugs:

Reply:
AssertEqualWriteError -> ignore Raw, AssertMatchesCommandError -> ignore Raw and Message.